pkg/api/dtos/plugins.go

```diff
@@ -28,7 +28,7 @@ type PluginSetting struct {
 	SignatureType   plugins.SignatureType   `json:"signatureType"`
 	SignatureOrg    string                  `json:"signatureOrg"`
 	AngularDetected bool                    `json:"angularDetected"`
-	APIVersion      string                  `json:"apiVersion"`
```

Comment:
the backend is not a single apiVersion -- it may have a few and a preferred version

Reply:
why not simplify this just using the `preferredVersion` here? In which case the frontend would be interested in knowing all the versions the backend supports?

Also, why duplicating this information? This is currently available in the endpoing `/apis` for the datasource backend endpoint.

Reply:
replied already here: https://github.com/grafana/grafana/pull/87639#discussion_r1606484446

Not sure if the frontend needs to handle this complexity (multiple api versions) but it's syntactically correct so if we want to move forward with this it can be okay

Reply:
Alternatively we can keep this out of the plugins DTO entirely -- I want to make sure we do not read it to mean more than it does.  The issue I have with the simplified version is that it encourages thinking about frontend+backend as one deployment rather than two things: client and server

Reply:
we need to have _something_ here if we want to set the value in queries / configs. The frontend will use the `preferredVersion` anyway and use it. I don't see a use case for the fronted to use the rest of the information so why adding it?

Reply:
What do we need to move this forward? Agreement on either only including preferred version or plus api group and versions? 

Given this is new/experimental can we start with just adding what we actually need right now and iterate figuring out the rest? As long as we don't officially support any new api response field we can add/remove/change how many times we want/need.

Reply:
As far as I understand from this comment: https://github.com/grafana/grafana/pull/87961#issuecomment-2124576258 we may want to just remove the `apiVersion` from the `/api/*` endpoints and leave it for the new ones under `/apis/*` :shrug: